### DISCUSSION TITLE
SentryPeer addition?

### DISCUSSION THREAD

**Discussion by ghenry** [telekom-security/tpotce#1017] [Ideas]
Hi all,

I couldn’t see a FAQ or guidance on how to submit a new HoneyPot?

I’m working on SentryPeer and it would be suitable at v1 to be used in your project.

https://github.com/SentryPeer/SentryPeer

Take a look and have a play? I’m currently working on the data sharing and replication features.

Thanks.

**Comment by t3chn0m4g3** [comment]
Thank you for the submission, I will play with it, hopefully even for 22.03 release.
Two questions right away, will the user be able to opt in / out of data sharing and does it have JSON logging to feed ES?

**Reply by ghenry** [reply]
Hi,

Data sharing will be opt in/out and they will have to share in order to receive other peer to peer data, but that’s not there in v1. I’m working on that now. v1 is a standalone SIP HoneyPot with RESTful API to check phone numbers and IPs.

Bad actors are logged to Syslog, IP address only at the moment for fail2ban. ES can consume syslog easily via LogStash if ELK.

I plan to expand the p2p sharing to other protocols, so there could be loads of users sharing honeypot data…

Thanks.

**Reply by t3chn0m4g3** [reply]
Thanks for the swift feedback, added this to the [board](https://github.com/telekom-security/tpotce/projects/5#card-78025522). A unified logfile would be a great thing to have (IPs, Phone No., etc.).

**Reply by ghenry** [reply]
Any feature requests, fire them over - https://github.com/SentryPeer/SentryPeer/issues/new?assignees=&labels=enhancement&template=feature_request.md&title=Enhancement%3A+

**Reply by t3chn0m4g3** [reply]
Thank you, very much appreciated!

**Reply by ghenry** [reply]
Sorry, forgot to clarify. A unified log file, do you mean the syslog version we've done is OK or that you'd still like JSON?

Thanks.

**Reply by t3chn0m4g3** [reply]
I do always prefer JSON logs over any other log file type as it saves so much time not to have to worry about field names with regard to ingestion, especially if you want to work with the individual fields for visualization purposes.

I mean ingestions like this ...
```
  if [type] == "Example1" {
    grok { match => { "message" => [ "\A%{TIMESTAMP_ISO8601:timestamp}%{SPACE}%{WORD:reason}%{SPACE}host=%{IPV4:src_ip}%{SPACE}port=%{INT:src_port}%{SPACE}fd=%{INT}%{SPACE}time=%{SECOND:duration}%{SPACE}bytes=%{NUMBER:bytes}", "\A%{TIMESTAMP_ISO8601:timestamp}%{SPACE}%{WORD:reason}%{SPACE}host=%{IPV4:src_ip}%{SPACE}port=%{INT:src_port}%{SPACE}fd=%{INT}%{SPACE}n=%{INT}/%{INT}" ] } }
    date {
      match => [ "timestamp", "ISO8601" ]
      remove_field => ["timestamp"]
    }
    mutate {
      add_field => {
        "dest_port" => "22"
        "dest_ip" => "${MY_EXTIP}"
      }
    }
  }
```

vs.

```
  if [type] == "Example2" {
    date {
      match => [ "timestamp", "ISO8601" ]
    }
    mutate {
      rename => {
        "dst_port" => "dest_port"
        "dst_ip" => "dest_ip"
      }
    }
  }
```

... makes all the difference, especially when fields change upstream, or new features are introduced resulting in a different `message`.

**Reply by ghenry** [reply]
Understood. So this would be json logged to syslog then shipped to logstash or a logfile or stdout etc for a container?

**Reply by t3chn0m4g3** [reply]
Having a unique JSON logfile just for SentryPeer would be perfect!

**Reply by ghenry** [reply]
Hmmm, OK. I'll think about a -j flag or similar to write lines of json to a stdout and/or a file? Is there a normal design pattern other project uses that I could checkout? Or a --logstash flag or similar so I can stick to a known technique

**Reply by t3chn0m4g3** [reply]
Your code is in C, so maybe [this](https://github.com/t3chn0m4g3/honeytrap/blob/master/src/modules/htm_logJSON.c) might help for starters. This was thankfully introduced to honeytrap by @adepasquale.

**Reply by ghenry** [reply]
Thanks. WIll take a look.

**Reply by ghenry** [reply]
Hi @t3chn0m4g3 

I've been working on this format for logging to a file enabled via a `-j` flag and `SENTRYPEER_JSON_LOG=1` for  json logging. I think it's a good idea for others too, so thanks for the suggestion. I've come up with this format:

```
{
   "app_name":"sentrypeer",
   "app_version":"v1.0.1",
   "event_timestamp":"2022-02-22 11:19:15.848934346",
   "event_uuid":"4503cc92-26cb-4b3e-bb33-69a83fa09321",
   "created_by_node_id":"4503cc92-26cb-4b3e-bb33-69a83fa09321",
   "collected_method":"responsive",
   "transport_type":"UDP",
   "source_ip":"45.134.144.128",
   "destination_ip":"XX.XX.XX.XX",
   "called_number":"0046812118532",
   "sip_method":"OPTIONS",
   "sip_user_agent":"friendly-scanner",
   "sip_message":"full SIP message"
}
```

Is this too much info? Should the `event_*` be first in the object and `app_*` last, or is this OK? I did a lot of research last night. These fields are all in the `bad_actor` data structure anyway, just a lot not exposed in syslog, just sqlite db and debug logs.

Thanks.

**Reply by ghenry** [reply]
I'll also let you set a location for the file, unless this should just go to `STDOUT`?

**Reply by t3chn0m4g3** [reply]
Awesome! Having the option of file location is great (`std_out` is turned off where possible), makes it easier to work with volumes / tmp in order to run the container in `read_only` mode.
Based on last night's tests, I think `User Agent` (attacker), `From` and `To` would be great to have, too, if possible.

**Reply by ghenry** [reply]
Good! 

`sip_user_agent` is in above JSON :-) 

I parse the "To" header in the SIP payload to grab the username/called number. I'm not interested in the From. It's in the full `sip_message` if a user is though. The `From` doesn't have anything of interest apart from the SIP syntax a bad actor is using. At least in the core use case of SentryPeer anyway.

Thanks.

**Reply by t3chn0m4g3** [reply]
Thanks! Gotcha 😄

**Reply by ghenry** [reply]
Hi @t3chn0m4g3 

Just about to commit. File contents per line is like:

```
cat sentrypeer_json.log 
{"app_name":"sentrypeer","app_version":"1.0.1","event_timestamp":"2022-02-22 22:09:41.187180857","event_uuid":"c86829ca-6016-4549-9f47-897da9407b1a","created_by_node_id":"c86829ca-6016-4549-9f47-897da9407b1a","collected_method":"passive","transport_type":"UDP","source_ip":"104.149.141.214","destination_ip":"8.8.8.8","called_number":"","sip_method":"","sip_user_agent":"","sip_message":""}
```

Is that usable one json object per line or do the whole file need to be a blob of json, i.e. a json array?

Thanks.

**Reply by t3chn0m4g3** [reply]
This is perfect 🤩  Thank you!

**Reply by ghenry** [reply]
https://github.com/SentryPeer/SentryPeer/commit/af4f88c1a780e72f686a53a8cc3e3bac1ab32cf8 :-) May need some more work for Zyre and OpenDHT detection, but the guts are in place now.

**Reply by t3chn0m4g3** [reply]
I started to adjust the `Dockerfile` for DHT and Zyre, but building costs are very high, especially WRT to ARM64 builds. In the hopes to reduce the building times I tried Debian and Ubuntu as alternatives as they seemed to have DHT as packaged lib (`libopendht-dev`), but turns out that does not work either as it only provides `/usr/include/opendht.h`:
```
tests/unit_tests/test_peer_to_peer_dht.c:26:10: fatal error: opendht/opendht_c.h: No such file or directory
   26 | #include <opendht/opendht_c.h>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: Leaving directory '/SentryPeer'
make[1]: *** [Makefile:1509: tests/unit_tests/runner-test_peer_to_peer_dht.o] Error 1
make: *** [Makefile:1988: check-am] Error 2
```

Maybe I am overlooking something, appreciate any thoughts :)

**Reply by ghenry** [reply]
Unfortunately `<opendht/opendht_c.h>` is only in their master branch at the moment. I've just done this for their latest release today https://github.com/Homebrew/homebrew-core/pull/95724 and they are soon to do a new release so we can pull it in via brew.

I can either:

1. Check for HAVE_OPENDHT_C in Makefile.am (am working on this already) and source code
2. Provide a --disable-opendht
3. Delete that code as it's not doing anything apart from running a test or two at the moment

I can go for 3 and tag a release, the put it back in main branch.

**Reply by t3chn0m4g3** [reply]
Thank you for the swift feedback!
I would need to test if homebrew is a viable option, going with `3.` now and have a switch `2.` for later would be perfect.

**Reply by ghenry** [reply]
I'm in the same place right now - https://github.com/SentryPeer/SentryPeer/runs/5295751673?check_suite_focus=true

Build it from source in GitHub Actions/Docker, or wait for release, or run own PPA, or use in Homebrew....

**Reply by t3chn0m4g3** [reply]
Not alone 😄

**Reply by ghenry** [reply]
Going to go with disable flag off the bat. BBL.

**Reply by ghenry** [reply]
```
./configure --help
...
Optional Features:
  --disable-option-checking  ignore unrecognized --enable/--with options
  --disable-FEATURE       do not include FEATURE (same as --enable-FEATURE=no)
  --enable-FEATURE[=ARG]  include FEATURE [ARG=yes]
  --enable-silent-rules   less verbose build output (undo: "make V=1")
  --disable-silent-rules  verbose build output (undo: "make V=0")
  --enable-dependency-tracking
                          do not reject slow dependency extractors
  --disable-dependency-tracking
                          speeds up one-time build
  --disable-opendht       Do not build with OpenDHT support
  --disable-zyre          Do not build with Zyre support
```

now to apply to source code.

**Reply by t3chn0m4g3** [reply]
Awesome. Thank you!

**Reply by ghenry** [reply]
All done. Will `git push` this evening "after work". Technically this is a side/personal project :-)

**Reply by ghenry** [reply]
Later I'll think about just calling it `--disable-p2p` as this is one of the deployment scenarios I want.

**Reply by t3chn0m4g3** [reply]
Perfect! And it seems we are in the same club 🤣

**Reply by ghenry** [reply]
Sneaky push!

**Reply by t3chn0m4g3** [reply]
Perfect!

I just noticed you seem to have the same problem with `libosip2` [here](https://github.com/SentryPeer/SentryPeer/runs/5295751673?check_suite_focus=true#step:4:87). My build shows me the same warning.

**Reply by t3chn0m4g3** [reply]
My build with `./configure --disable-opendht --disable-zyre` still wants `zyre.h` to be present.

```
Step 7/17 : RUN ./configure --disable-opendht --disable-zyre
 ---> Running in 41038d46dab6
checking build system type... x86_64-pc-linux-musl
checking host system type... x86_64-pc-linux-musl
checking target system type... x86_64-pc-linux-musl
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether the compiler supports GNU C... yes
checking whether gcc accepts -g... yes
checking for gcc option to enable C11 features... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for stdio.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for strings.h... yes
checking for sys/stat.h... yes
checking for sys/types.h... yes
checking for unistd.h... yes
checking for wchar.h... yes
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking whether _XOPEN_SOURCE should be defined... no
checking for g++... g++
checking whether the compiler supports GNU C++... yes
checking whether g++ accepts -g... yes
checking for g++ option to enable C++11 features... none needed
checking dependency style of g++... gcc3
checking for gawk... (cached) awk
checking for gcc... (cached) gcc
checking whether the compiler supports GNU C... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to enable C11 features... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) gcc3
checking how to run the C preprocessor... gcc -E
checking whether ln -s works... yes
checking whether make sets $(MAKE)... (cached) yes
checking for ranlib... ranlib
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for libosip2... no
configure: WARNING: It is highly recommended to install libosip2 >= 5.0
checking for library containing osip_message_init... -losipparser2
checking for library containing sqlite3_open... -lsqlite3
checking for library containing uuid_generate... -luuid
checking for library containing json_object... -ljansson
checking for library containing curl_easy_init... -lcurl
checking for library containing MHD_start_daemon... -lmicrohttpd
checking for library containing pcre2_compile_8... -lpcre2-8
checking for library containing pthread_create... none required
checking for arpa/inet.h... yes
checking for netdb.h... yes
checking for netinet/in.h... yes
checking for stddef.h... yes
checking for stdint.h... (cached) yes
checking for stdlib.h... (cached) yes
checking for string.h... (cached) yes
checking for strings.h... (cached) yes
checking for sys/socket.h... yes
checking for sys/time.h... yes
checking for syslog.h... yes
checking for unistd.h... (cached) yes
checking for signal.h... yes
checking for pthread.h... yes
checking for _Bool... yes
checking for stdbool.h that conforms to C99... yes
checking for int64_t... yes
checking for size_t... yes
checking for uint32_t... yes
checking for GNU libc compatible malloc... yes
checking for GNU libc compatible realloc... yes
checking for working strnlen... yes
checking for setenv... yes
checking for clock_gettime... yes
checking for getcwd... yes
checking for inet_ntoa... yes
checking for localtime_r... yes
checking for memset... yes
checking for select... yes
checking for socket... yes
checking for strdup... yes
checking for strncasecmp... yes
checking for strndup... yes
checking for atexit... yes
checking for bzero... yes
checking for a sed that does not truncate output... /bin/sed
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking whether gcc is Clang... no
checking whether pthreads work with "-pthread" and "-lpthread"... yes
checking for joinable pthread attribute... PTHREAD_CREATE_JOINABLE
checking whether more special flags are required for pthreads... no
checking for PTHREAD_PRIO_INHERIT... yes
checking for OPENDHT_C... no
configure: WARNING: OpenDHT-c is not detected via pkg-config. Please install it manually.
checking for ZYRE... no
configure: WARNING: zyre is not detected via pkg-config. Please install it manually.
checking for CZMQ... no
configure: WARNING: libczmq is not detected via pkg-config. Please install it manually.
checking for ZMQ... no
configure: WARNING: libzmq is not detected via pkg-config. Please install it manually.
checking for git... 1.7.0
checking for IP_PKTINFO... yes
checking for IP_RECVDSTADDR... no
checking if you'd like to disable OpenDHT support... yes
checking if you'd like to disable Zyre support... yes
checking for doxygen... no
configure: WARNING: Doxygen not found - continuing without Doxygen support
checking for doxygen... no
configure: WARNING: Doxygen not found - continuing without Doxygen support
checking for stdarg.h... yes
checking for stddef.h... (cached) yes
checking for setjmp.h... yes
checking for CMOCKA... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating config.h
config.status: executing depfiles commands
configure:
Configured build options for sentrypeer:
  Configure date:     2022-02-23 17:03
  Build arch:         x86_64-pc-linux-musl
  Build user:
  Build host:         41038d46dab6
  Install prefix:     /usr/local
  Compiler:
   C compiler         gcc
   CFLAGS:            -g -O2 -pthread
   LIBS:              -lpthread -lpcre2-8 -lmicrohttpd -lcurl -ljansson -luuid -lsqlite3 -losipparser2

  Configure complete! Now proceed with:
  - 'make'            compile sentrypeer
  - 'make check'      run the sentrypeer self tests
  - 'make install'    install the 'sentrypeer' project to /usr/local

Removing intermediate container 41038d46dab6
 ---> f7fa676a001e
Step 8/17 : RUN make
 ---> Running in 86837122393a
make  all-am
make[1]: Entering directory '/SentryPeer'
depbase=`echo src/sentrypeer.o | sed 's|[^/]*$|.deps/&|;s|\.o$||'`;\
gcc -DHAVE_CONFIG_H -I.    -Wall -Werror -Wextra -Wpedantic -pedantic -Wformat=2 -Wno-unused-parameter -Wshadow -Wwrite-strings -Wstrict-prototypes -Wold-style-definition -Wnested-externs -Wmissing-include-dirs -std=c18 -D_FORTIFY_SOURCE=2 -fpie -fpic -g3 -O2 -fstack-protector-strong -grecord-gcc-switches -Werror=format-security -Werror=implicit-function-declaration -Wmisleading-indentation  -g -O2 -pthread -MT src/sentrypeer.o -MD -MP -MF $depbase.Tpo -c -o src/sentrypeer.o src/sentrypeer.c &&\
mv -f $depbase.Tpo $depbase.Po
In file included from src/sentrypeer.c:25:
src/conf.h:20:10: fatal error: zyre.h: No such file or directory
   20 | #include <zyre.h>
      |          ^~~~~~~~
compilation terminated.
make[1]: Leaving directory '/SentryPeer'
make[1]: *** [Makefile:1116: src/sentrypeer.o] Error 1
make: *** [Makefile:823: all] Error 2
```

**Reply by ghenry** [reply]
Thanks. Just working on it now.

**Reply by ghenry** [reply]
This should be all working now. I've been chasing a GitHub Action red herring for ages today/tonight. The make `build_and_test` has been passing, but `make distcheck` runs a plain `./configure` and I'm not using `HAVE_ZYRE` or `HAVE_OPENDHT` to check for includes etc. Arghhh. I'll sort that tomorrow, but you should be good now is using `main`.

I didn't scroll down on the results! Been using this to test locally - https://github.com/nektos/act Very nice!!! 

Thanks.

**Reply by t3chn0m4g3** [reply]
Thanks, on it now. TTYL

**Reply by t3chn0m4g3** [reply]
Works as advertised 😃 
```
{
  "app_name": "sentrypeer",
  "app_version": "1.0.1",
  "event_timestamp": "2022-02-23 22:11:50.776640982",
  "event_uuid": "dbf16d72-16c4-46fa-b897-863804bbab92",
  "created_by_node_id": "dbf16d72-16c4-46fa-b897-863804bbab92",
  "collected_method": "responsive",
  "transport_type": "UDP",
  "source_ip": "172.20.254.190",
  "destination_ip": "172.26.0.2",
  "called_number": "john",
  "sip_method": "REGISTER",
  "sip_user_agent": "Telephone 1.5.2",
  "sip_message": "REGISTER sip:172.20.254.102 SIP/2.0\r\nVia: SIP/2.0/UDP 172.20.254.190:62533;rport;branch=z9hG4bKPjfceo9WR9VToeMFRZQG1poZhUehWz1cTC\r\nFrom: \"john\" <sip:john@172.20.254.102>;tag=.NcE0pMdOUA-X3oKzGeHFY36PlaHADSL\r\nTo: \"john\" <sip:john@172.20.254.102>\r\nCall-ID: bi.sbkm1J4fdbpyChfhcmc3S1ZxCW-yX\r\nCSeq: 4429 REGISTER\r\nContact: \"john\" <sip:john@172.20.254.190:62533;ob>\r\nAllow: PRACK\r\nAllow: INVITE\r\nAllow: ACK\r\nAllow: BYE\r\nAllow: CANCEL\r\nAllow: UPDATE\r\nAllow: INFO\r\nAllow: SUBSCRIBE\r\nAllow: NOTIFY\r\nAllow: REFER\r\nAllow: MESSAGE\r\nAllow: OPTIONS\r\nMax-forwards: 70\r\nUser-agent: Telephone 1.5.2\r\nExpires: 300\r\nContent-Length: 0\r\n\r\n"
}
```

Will ping you, once integration with ELK is finalized.

Thank you very much for the great collaboration!

**Reply by ghenry** [reply]
Ding Ding!!! Nice work! I've met, worked and connected with tons of great people since starting SentryPeer. Side projects rule!

Keep up the great work and let me know if there's anything else you'd like to see.

Thanks again!

**Reply by ghenry** [reply]
There might be a funny smell with the timestamp, let me know how you get on parsing it in ELK. I stick nanoseconds on for when SentryPeer gets hammered by probes:

https://github.com/SentryPeer/SentryPeer/blob/main/src/utils.c#L25

**Reply by ghenry** [reply]
> Perfect!
> 
> I just noticed you seem to have the same problem with `libosip2` [here](https://github.com/SentryPeer/SentryPeer/runs/5295751673?check_suite_focus=true#step:4:87). My build shows me the same warning.

This should be gone now too.

**Reply by t3chn0m4g3** [reply]
Awesome, works just fine and likewise 😃

**Comment by ghenry** [comment]
Are you guys all from https://www.telekom.com/ then?

**Reply by t3chn0m4g3** [reply]
Yes, from the [Deutsche Telekom Security GmbH](https://www.t-systems.com/de/en/security) unit.

**Reply by ghenry** [reply]
Very cool :-)

**Comment by t3chn0m4g3** [comment]
I just played around a little and noticed the following:
```
sentrypeer    | Source IP: 172.27.0.1
sentrypeer    | Called Number: .oO.oO
sentrypeer    | SIP Method: REGISTER
sentrypeer    | Transport Type: UDP
sentrypeer    | User Agent: Linphone Desktop/4.3.2 (macOS 10.16, Qt 5.15.2) LinphoneCore/5.0.49
sentrypeer    | Collected Method: responsive
sentrypeer    | Created by Node Id: 4753e0ad-c760-4c60-94cb-55723c28ea3d
sentrypeer    | SentryPeer db file location is: sentrypeer.db
sentrypeer    | Destination IP address of UDP packet is: 172.27.0.2
sentrypeer    | epochtime: 1645484236
sentrypeer    | Received (544 bytes): REGISTER sip:172.20.254.102 SIP/2.0
sentrypeer    | Via: SIP/2.0/UDP 172.20.254.190:56475;rport;branch=z9hG4bKPjJkPF7ao3MJ01hmE.291uVCqgNlq-y78a
sentrypeer    | Max-Forwards: 70
sentrypeer    | From: "john" <sip:john@172.20.254.102>;tag=LhdIkBaASo4qQL7US617wS7m1TToEqMA
sentrypeer    | To: "john" <sip:john@172.20.254.102>
sentrypeer    | Call-ID: XIcogmOV01Baa0JHOJ1KB6T.hqpw2JsG
sentrypeer    | CSeq: 42804 REGISTER
sentrypeer    | User-Agent: Telephone 1.5.2
sentrypeer    | Contact: "john" <sip:john@172.20.254.190:56475;ob>
sentrypeer    | Expires: 300
sentrypeer    | Allow: PRACK, INVITE, ACK, BYE, CANCEL, UPDATE, INFO, SUBSCRIBE, NOTIFY, REFER, MESSAGE, OPTIONS
sentrypeer    | Content-Length:  0
sentrypeer    |
sentrypeer    |
sentrypeer    | read_packet_buf size is: 1024:
sentrypeer    | read_packet_buf length is: 544:
sentrypeer    | bytes_received size is: 544:
sentrypeer    |
sentrypeer    | Bad Actor is:
sentrypeer    | Event Timestamp: 2022-02-21 22:57:16.303844711
sentrypeer    | Event UUID: 71a122f7-d17c-4db9-a3db-88a08b00b171
sentrypeer    | SIP Message: REGISTER sip:172.20.254.102 SIP/2.0
sentrypeer    | Via: SIP/2.0/UDP 172.20.254.190:56475;rport;branch=z9hG4bKPjJkPF7ao3MJ01hmE.291uVCqgNlq-y78a
sentrypeer    | From: "john" <sip:john@172.20.254.102>;tag=LhdIkBaASo4qQL7US617wS7m1TToEqMA
sentrypeer    | To: "john" <sip:john@172.20.254.102>
sentrypeer    | Call-ID: XIcogmOV01Baa0JHOJ1KB6T.hqpw2JsG
sentrypeer    | CSeq: 42804 REGISTER
sentrypeer    | Contact: "john" <sip:john@172.20.254.190:56475;ob>
sentrypeer    | Allow: PRACK
sentrypeer    | Allow: INVITE
sentrypeer    | Allow: ACK
sentrypeer    | Allow: BYE
sentrypeer    | Allow: CANCEL
sentrypeer    | Allow: UPDATE
sentrypeer    | Allow: INFO
sentrypeer    | Allow: SUBSCRIBE
sentrypeer    | Allow: NOTIFY
sentrypeer    | Allow: REFER
sentrypeer    | Allow: MESSAGE
sentrypeer    | Allow: OPTIONS
sentrypeer    | Max-forwards: 70
sentrypeer    | User-agent: Telephone 1.5.2
sentrypeer    | Expires: 300
sentrypeer    | Content-Length: 0
```
The logs contain anything you might wish for, but the source IP will currently only be reflected correctly if the container's network is running in host mode, which I want to avoid unless completely necessary. While this is typical for Docker with regard to the destination IP I was not aware this might be an issue with the source IP.

You can reproduce with the following `docker-compose.yml`:
```
version: '2.3'

networks:
  sentrypeer_local:

services:

# SentryPeer service
  sentrypeer:
    build: .
    container_name: sentrypeer
    restart: always
    networks:
     - sentrypeer_local
    #network_mode: "host"
    ports:
     - "5060:5060/udp"
     - "8082:8082"
    image: "dtagdevsec/sentrypeer:2203"
```
I used the `Dockerfile` provided from the `v1.0.0` with the adjustment of `RUN git clone https://github.com/SentryPeer/SentryPeer.git -b v1.0.0` (or it would not build due to missing `Zyre.h`).

**Comment by ghenry** [comment]
Thanks for testing!

I’ll look at the Dockerfile. I plan to clean it up more as it’s including the source files.

See last comment on https://github.com/moby/libnetwork/issues/1994

Does that work? This is a Docker issue as I’m using what ever the source of the packet is at the lowest level possible.

**Reply by t3chn0m4g3** [reply]
Thanks, correct, re-checked this with other UDP based services, same result. Will test `conntrack -D -p udp` next.

**Reply by t3chn0m4g3** [reply]
Confirmed working! Thx!

**Reply by ghenry** [reply]
Awesome! That should help with the other projects too?

**Reply by t3chn0m4g3** [reply]
Yes, with ddospot which will be introduced in 22.03.

**Comment by ghenry** [comment]
Thanks. It's a real PITA for everyone.

**Comment by t3chn0m4g3** [comment]
So, [this](https://github.com/telekom-security/tpotce/tree/22.x/docker/sentrypeer) works 😄

**Reply by ghenry** [reply]
Awesome work! Can I pretty please steal those? 

Thank you for all of this. 

PS. We don't do TCP yet for SIP.

**Reply by t3chn0m4g3** [reply]
Of course 😊

Thanks for the great work and collaboration.

**Reply by ghenry** [reply]
My pleasure. Teamwork!

**Comment by t3chn0m4g3** [comment]
Dashboard is done (artificial attack data) and integration will be finalized soon.
![image](https://user-images.githubusercontent.com/4318452/155717767-08a7ac18-4dfb-41e9-9da4-0752a9fb7d57.png)

**Comment by ghenry** [comment]
Amazing! Saves me so much work too. Is it possible to search for SentryPeer
in that case?

**Reply by t3chn0m4g3** [reply]
Yes, all data is indexed as `type:"Sentrypeer"`.

**Reply by ghenry** [reply]
Awesome! Does SentryPeer make SIP easier to manage than others you've used then? I think it's Dionaea?

Will this data show up at https://www.sicherheitstacho.eu/start/main ?

**Reply by t3chn0m4g3** [reply]
Once the integration in `ewsposter` is done by @trixam and T-Pot 22.03 is released it should be shown in Sicherheitstacho.

**Reply by ghenry** [reply]
Cool. I''ve clicked Watch on the main project. I'm going to do a v1.2.0 tag tonight so you could use that for the json instead of `main`

**Reply by t3chn0m4g3** [reply]
Perfect! Thank you!

**Comment by ghenry** [comment]
Hi @t3chn0m4g3 

Is [this line](https://github.com/telekom-security/tpotce/blob/22.x/docker/sentrypeer/Dockerfile#L56) working for you in your Dockerfile? 

` chown -R sentrypeer:sentrypeer /opt/sentrypeer && \`

Don't you also need /var/log to be owned by the sentrypeer user?

**Reply by ghenry** [reply]
Nevermind. Was reading something wrongly. All good.

**Reply by ghenry** [reply]
Yeah, actually I had to change to:

```
    chown -R sentrypeer:sentrypeer /opt/sentrypeer /var/lib/sentrypeer /var/log/sentrypeer && \
```

**Reply by t3chn0m4g3** [reply]
This is happening on the T-Pot host by using volumes and owning the path. You can see that in the installer and the docker-compose.yml.

**Reply by ghenry** [reply]
Yeah, thought that after I tested my own Dockerfile. Ignore me :-)

**Reply by t3chn0m4g3** [reply]
Been there, done that 🤪

**Reply by ghenry** [reply]
AND I forgot 5060:5060/udp !!!

**Comment by ghenry** [comment]
Might be handy for you @t3chn0m4g3 https://github.com/SentryPeer/SentryPeer/commit/dd5bc4d0ffd26594af8ed09b757899b8fb2c07ce

**Reply by t3chn0m4g3** [reply]
Thanks!
